module/zfs/mmp.c

```diff
@@ -330,13 +370,25 @@ mmp_write_uberblock(spa_t *spa)
 		zfs_dbgmsg("SCL_STATE acquisition took %llu ns\n",
 		    (u_longlong_t)lock_acquire_time);
 
-	vd = mmp_random_leaf(spa->spa_root_vdev);
+	vd = mmp_random_leaf(spa->spa_root_vdev, &error);
+
+	mutex_enter(&mmp->mmp_io_lock);
+
 	if (vd == NULL) {
```

Comment:
Can you add a brief comment before this block describing the skipped history entries.

Reply:
done